sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs

```diff
@@ -178,6 +173,167 @@ impl Wallet {
         // store new data
         #[cfg(feature = "storage")]
         {
+            use crate::wallet::wallet::operations::storage::SaveLoadWallet;
+            let wallet_builder = WalletBuilder::new()
+                .with_secret_manager_arc(self.secret_manager.clone())
+                .with_storage_path(
+                    &self
+                        .storage_options
+                        .storage_path
+                        .clone()
+                        .into_os_string()
+                        .into_string()
+                        .expect("can't convert os string"),
+                )
+                .with_client_options(self.client_options().await)
+                .with_coin_type(self.coin_type.load(Ordering::Relaxed));
+            // drop secret manager, otherwise we get a deadlock in save_wallet_data
+            drop(secret_manager);
+            wallet_builder.save_data(&*self.storage_manager.read().await).await?;
+            // also save account to db
+            for account in accounts.iter() {
+                account.save(None).await?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Wallet<StrongholdSecretManager> {
```

Comment:
Same here, seems to work just fine with this whole block removed

Reply:
Same answer as above.